### PR TITLE
Fix not to overwrite Content-Type when the given :content is a string on Windows

### LINKED ISSUE #135: Windows: Content-Type is replaced with "text/plain" when `:content` is present
Forms to reproduce on SBCL on Windows:
```lisp
(require :dexador)
;; Tested with both :method :get and :method :post
;; httpbin responds with a JSON payload of the request data
(dexador:request "https://httpbin.org/anything" :method :get 
                 :headers '((:content-type "application/json")) :content "{}")
```

If *content* is present with a string, then the request is sent with the header `Content-Type: text/plain`, even though this is specified to be `Content-Type: application/json`. When *content* is removed, the request is sent with the header `Content-Type: application/json`.

I've found this bug on Windows. On Linux, both forms send a request with header `Content-Type: application/json` as expected.

Just a note on reproducing this bug, I had to apply this PR https://github.com/usocket/usocket/pull/89 to Quicklisp's latest version of usocket; otherwise, it would fail to compile on Windows.

Thanks!

### PR DESCRIPTION
**Pull request by fukamachi**
fixes #135

### CODE REVIEW

**Comment by ajberkley** [comment]
This looks OK to me, are you waiting for something before merging this?

**Comment by mattboroi** [comment]
while using this branch on windows 10 (22h2) with sbcl 2.2.5 the following experience was observed.

`* (dex:post "https://httpbin.org/anything" :headers '(("content-type" . "application/json")) :content "{}")`

```
Heap exhausted during allocation: 242843648 bytes available, 335544336 requested.
 Gen  Boxed   Cons    Raw   Code  SmMix  Mixed  LgRaw LgCode  LgMix Waste%       Alloc        Trig   Dirty GCs Mem-age
  0    1231    466      0      0      0    153      0      0      0    0.5    60331952    64533370       -   1  0.0000
  1       0      0      0      0      0      0      0      0      0    0.0           0    10737418       0   0  0.0000
  2    4572    126    181    147     50     37  10256      0     40    0.3   503240000     2000000   10438   0  0.0000
  3       0      0      0      0      0      0      0      0      0    0.0           0     2000000       0   0  0.0000
  4       0      0      0      0      0      0      0      0      0    0.0           0     2000000       0   0  0.0000
  5       0      0      0      0      0      0      0      0      0    0.0           0     2000000       0   0  0.0000
  6     332    120    186    353     67     15     10      0     55    2.2    36472992     2000000      11   0  0.0000
Tot    6135    712    367    500    117    205  10266      0     95    0.5   600044944 [55.9% of 1073741824 max]
GC control variables:
   *GC-INHIBIT* = false
   *GC-PENDING* = true
   *STOP-FOR-GC-PENDING* = false

debugger invoked on a SB-KERNEL::HEAP-EXHAUSTED-ERROR in thread
#<THREAD "main thread" RUNNING {1000608003}>:
  Heap exhausted (no more space for allocation).
242843648 bytes available, 335544336 requested.

PROCEED WITH CAUTION.

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [ABORT] Exit debugger, returning to top level.

(SB-KERNEL::HEAP-EXHAUSTED-ERROR 121421824 167772168)
```

**Comment by mattboroi** [comment]
[0001-hack-to-handle-specified-content-type.patch.txt](https://github.com/fukamachi/dexador/files/10047748/0001-hack-to-handle-specified-content-type.patch.txt)

**Comment by ajberkley** [comment]
Your patch is just a bugfix to Fukamachi's patch where he forgot a (cdr content-type) when calling convert-content?

**Comment by mattboroi** [comment]
@ajberkley that's correct.

should have spent an extra two minutes to review both to see the obvious, doh.
the attached patch above was a local change i had been using before finding this pr.

File: src/backend/winhttp.lisp [hunk 1]
```diff
@@ -119,7 +130,7 @@
          (user-agent
            (cdr (find :user-agent headers :key #'car :test #'string-equal))))
     (multiple-value-bind (content detected-content-type)
-        (convert-content content multipart-p form-urlencoded-p)
+        (convert-content content multipart-p form-urlencoded-p content-type)
```
Comment (mattboroi) [inline src/backend/winhttp.lisp:133]: changing to `(cdr content-type)` resolves an observed heap exhaustion error during allocation.

Comment (mattboroi) [reply src/backend/winhttp.lisp:133]: no perms to push that change but otherwise would.

**Comment by ajberkley** [comment]
@mattboroi was using this successfully and it happened to overlap with @fukamachi 's solution so I fixed the cdr problem in the latter and merged it.

**Pull request merged**